### PR TITLE
Provide a BaseApp for QtWebEngine 5.13

### PR DESCRIPTION
**Pull request by aleixpol**
This is the patch to adapt Falkon:
https://phabricator.kde.org/P502

### CODE REVIEW

File: io.qt.qtwebengine.BaseApp.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    "runtime-version" : "5.13",
+    "modules" : [
+        {
+            "name" : "qt5-qtwebengine",
+            "buildsystem" : "qmake",
+            "cleanup-platform" : [
+                "/bin",
+                "/mkspecs"
+            ],
+            "make-install-args": ["INSTALL_ROOT=/app/potatoes"],
```
Comment (barthalion) [inline io.qt.qtwebengine.BaseApp.json]: Uh, is it really what you want?

File: io.qt.qtwebengine.BaseApp.json [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                "-webengine-proprietary-codecs",
+                "-webengine-icu",
+                "-webp"
+            ],
+            "post-install" : [
+                "cp -r /app/potatoes/usr/* /app",
+                "rm -rf /app/potatoes"
+            ],
+            "modules" : [
+                "shared-modules/python2.7/python-2.7.json",
```
Comment (barthalion) [inline io.qt.qtwebengine.BaseApp.json]: If it's needed only build time, copy-paste entire module and add `"cleanup": ["*"]`

File: io.qt.qtwebengine.BaseApp.json [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                    "sha256" : "adcf56b5de6f34744bba2307b696fc75611884f4270e87dfa760d6e99dd711bb"
+                },
+                {
+                    "type" : "patch",
+                    "path" : "patches/resources.patch"
+                }
+            ]
+        },
+        {
+            "name": "polish",
```
Comment (barthalion) [inline io.qt.qtwebengine.BaseApp.json]: I would prefer to have it done in `post-install`.

File: io.qt.qtwebengine.BaseApp.json [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    "id" : "io.qt.qtwebengine.BaseApp",
+    "branch": "5.13",
+    "runtime" : "org.kde.Platform",
+    "sdk" : "org.kde.Sdk",
+    "runtime-version" : "5.13",
+    "modules" : [
+        {
+            "name" : "qt5-qtwebengine",
+            "buildsystem" : "qmake",
+            "cleanup-platform" : [
```
Comment (barthalion) [inline io.qt.qtwebengine.BaseApp.json]: Just `cleanup`.

**Comment by barthalion** [comment]
bot, build io.qt.qtwebengine.BaseApp

File: io.qt.qtwebengine.BaseApp.json (hunk 1 above)
Comment (aleixpol) [reply io.qt.qtwebengine.BaseApp.json]: Define "want"

File: io.qt.qtwebengine.BaseApp.json (hunk 3 above)
Comment (aleixpol) [reply io.qt.qtwebengine.BaseApp.json]: Why? I did it this way so it's easier to modify these hacks without having to rebuild all of qtwebengine every time.

**Comment by aleixpol** [comment]
bot, build io.qt.qtwebengine.BaseApp

**Comment by aleixpol** [comment]
bot, build io.qt.qtwebengine.BaseApp

**Comment by aleixpol** [comment]
bot, build io.qt.qtwebengine.BaseApp

File: io.qt.qtwebengine.BaseApp.json (hunk 1 above)
Comment (Erick555) [reply io.qt.qtwebengine.BaseApp.json]: I guess the question is if you really want to install into dir called `potatoes` and why.

File: io.qt.qtwebengine.BaseApp.json [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                "-webp"
+            ],
+            "post-install" : [
+                "cp -r /app/potatoes/usr/* /app",
+                "rm -rf /app/potatoes"
+            ],
+            "modules" : [
+                "shared-modules/python2.7/python-2.7.json",
+                {
+                    "name" : "pciutils",
```
Comment (Erick555) [inline io.qt.qtwebengine.BaseApp.json]: pciutils is in Platform/Sdk. You can dropit

File: io.qt.qtwebengine.BaseApp.json [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                        {
+                            "type" : "shell",
+                            "commands" : [
+                                "sed -i 's@PREFIX=/usr/local@PREFIX=/app@' Makefile"
+                            ]
+                        }
+                    ]
+                },
+                {
+                    "name" : "opus",
```
Comment (Erick555) [inline io.qt.qtwebengine.BaseApp.json]: opus is also in Platform/Sdk, drop it.

File: io.qt.qtwebengine.BaseApp.json (hunk 3 above)
Comment (Erick555) [reply io.qt.qtwebengine.BaseApp.json]: I guess the question is if you really want to install into dir called `potatoes` and why.

File: io.qt.qtwebengine.BaseApp.json (hunk 1 above)
Comment (Erick555) [reply io.qt.qtwebengine.BaseApp.json]: I guess the question is if you really want to install into dir called `potatoes` and why.

**Comment by aleixpol** [comment]
bot, build io.qt.qtwebengine.BaseApp

**Comment by aleixpol** [comment]
bot, build io.qt.qtwebengine.BaseApp

**Comment by Erick555** [comment]
@aleixpol for pciutils from Sdk this may need to be fixed first: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/merge_requests/2023

**Comment by aleixpol** [comment]
bot, build io.qt.qtwebengine.BaseApp

File: io.qt.qtwebengine.BaseApp.json (hunk 1 above)
Comment (aleixpol) [reply io.qt.qtwebengine.BaseApp.json]: It insists to install to /usr, it's hardcoded in Qt and I don't see a way around it. We need to install elsewhere and then move it into /app and place it properly. Hence using this name. I can change it if it bothers you.

**Comment by aleixpol** [comment]
bot, build io.qt.qtwebengine.BaseApp

**Comment by aleixpol** [comment]
bot, build io.qt.qtwebengine.BaseApp

**Comment by aleixpol** [comment]
bot, build io.qt.qtwebengine.BaseApp

File: io.qt.qtwebengine.BaseApp.json (hunk 3 above)
Comment (barthalion) [reply io.qt.qtwebengine.BaseApp.json]: So be it, I guess >40 minutes of build can justify that.

File: io.qt.qtwebengine.BaseApp.json (hunk 1 above)
Comment (barthalion) [reply io.qt.qtwebengine.BaseApp.json]: Yeah, calling it `scratchdir` would be better. BaseApps are often used as templates later.

**Comment by aleixpol** [comment]
bot, build io.qt.qtwebengine.BaseApp

**Pull request closed without merging**